### PR TITLE
refactor RadixTree

### PR DESCRIPTION
**Pull request by tigercosmos**
The `TimedEntry` struct located in `RadixTree.hpp` is intended for testing purposes. Let's relocate the struct to the test file.

### CODE REVIEW

**Comment by tigercosmos** [comment]
cc @q40603  @yungyuc

File: gtests/test_nopython_radixtree.cpp [hunk 1]
```diff
@@ -1,10 +1,37 @@
...
+#include <modmesh/toggle/RadixTree.hpp>
 
 #ifdef Py_PYTHON_H
 #error "Python.h should not be included."
 #endif
 
+/**
+ * Simple Timed Entry for Testing
+ */
+class TimedEntry
```
Comment (yungyuc) [inline gtests/test_nopython_radixtree.cpp]: Moving the test code here makes sense to me.  What do you think, @q40603 ?

We have another `TimedEntry` at https://github.com/solvcon/modmesh/blob/d6ed9ac95f449ff77eb6e4d379c83d25f60eac23/cpp/modmesh/toggle/profile.hpp#L102 

For the test code here, what about renaming it to be different from `modmesh::TimedEntry` there?

**Review by yungyuc — CHANGES_REQUESTED** [review]
- [x] Rename the class for the test code to be like test code.

File: gtests/test_nopython_radixtree.cpp (hunk 1 above)
Comment (tigercosmos) [reply gtests/test_nopython_radixtree.cpp]: renamed.

**Review by yungyuc — APPROVED** [review]
LGTM.  Wait for @q40603 comments

**Comment by q40603** [comment]
@tigercosmos 
Sorry for the late response.
Separating the TimedEntry class from Radix Tree is more elegant, and it also makes more sense as Radixtree should accept other objects as well.

**Comment by yungyuc** [comment]
Great!  Thank you @tigercosmos and @q40603

**Pull request merged**